[Demo] Add info about React Native app (#5704)
open-telemetry/opentelemetry.io: content/en/docs/demo/services/react-native-app.md

---
title: React Native App
cSpell:ignore: typeof
---

The React Native app provides a mobile UI for users on Android and iOS devices
to interact with the demo's services. It is built with
[Expo](https://docs.expo.dev/get-started/introduction/) and uses Expo's
file-based routing to layout the screens for the app.

[React Native app source](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/react-native-app/)

## Instrumentation

The application uses the OpenTelemetry packages to instrument the application at
the JS layer

{{% alert title="Important" color="warning" %}}

> The JS OTel packages are supported for node and web environments. While they
> work for React Native as well they are not currently explicitly supported for
> that environment and may break compatibility with minor version updates or
> require workarounds. Building more support for React Native is an area of active
> development.

Reviewer: ```suggestion
The JS OTel packages are supported for node and web environments. While they
work for React Native as well, they are not explicitly supported for
that environment, where they might break compatibility with minor version updates or
require workarounds. Building JS Otel package support for React Native is an area of active
development.
```
